R/stan_module.R

```diff
@@ -31,7 +31,18 @@ StanModule <- setClass(
 #' file in the package directory or the stan code as a string.
 #' @export
 read_stan <- function(string) {
-    system_file <- system.file("LogLogistic", string, package = "jmpost")
+    file_map <- list.files("inst/")
+    system_file <- apply(as.matrix(file_map),
+        1,
+        FUN = system.file,
+        package = "jmpost",
+        string = string
+    )
+    system_file <- system_file[which(nchar(system_file) != 0)]
+    if (length(system_file) == 0 || length(system_file) > 1 ) system_file <- ""
```

Comment:
why in the case of more than one file found we don't use them?

Reply:
It refers mainly to the case where the user leaves a section empty (eg the model section here `StanModule(functions = "file.stan", model = "")`). In that case, for the model section, the call is `read_stan("")` and therefore at some point inside the `read_stan` function we have the call `system_file( "folder_name", "", package = "jmpost")`.  The last call gives back the path for the folder. Since we have 3 folders inside the `inst` directory, we get back 3 characters with the 3 different paths.